collection-scripts/gather

```diff
@@ -1,40 +1,24 @@
 #!/bin/bash
+#
+# This is the main script that is invoked when we call `oc adm must-gather`,
+# and it is called directly regardless of the ENTRYPOINT in the Dockerfile
+#
+# This file is necessary to pipe the output of the must-gather process to the
+# output directory.
 
-DIR_NAME=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-# shellcheck disable=SC1091
-source "${DIR_NAME}/common.sh"
+LOG_DIR="${BASE_COLLECTION_PATH:-/must-gather}"
 
-# get apiservices
-/usr/bin/gather_apiservices
+# If directory doesn't exists it's because we are manually creating the
+# container for development, so we'll use a temporary directory
+if [[ ! -d "$LOG_DIR" ]]; then
+  LOG_DIR=`mktemp -d`
+  export BASE_COLLECTION_PATH="$LOG_DIR"
+  OC_CALLED=false
+fi
 
-# get CRDs
-/usr/bin/gather_crds
+LOG="${LOG_DIR}/run.log"
 
-# get CRs
-/usr/bin/gather_crs
+/usr/bin/gather_run 2>&1 | tee $LOG
```

Comment:
I see a few `shellcheck` related warnings, which might be good to analyze at this stage and then move forward. For instance here we have:

```

In collection-scripts/gather line 21:
/usr/bin/gather_run 2>&1 | tee $LOG
                               ^--^ SC2086 (info): Double quote to prevent globbing and word splitting.
```
that is just a matter of double quoting the variable.
Not a blocker though, I can address styling / cosmetic issues later (in a follow up) because there are a few

Reply:
Done